cmake/BDWGc.cmake

```diff
@@ -0,0 +1,40 @@
+macro(p4c_obtain_bdwgc)
+  include(CheckSymbolExists)
+  # Turn on C++ support. We do not need the headers and documentation.
+  set(enable_cplusplus ON CACHE BOOL "C++ support")
+  set(install_headers ON CACHE BOOL "Install header and pkg-config metadata files.")
+  set(enable_docs ${ENABLE_DOCS} CACHE BOOL "Build and install documentation.")
+  # We may have to compile very large programs which allocate a lot on the heap.
+  set(enable_large_config ON CACHE BOOL "Optimize for large heap or root set.")
+  set(without_libatomic_ops OFF CACHE BOOL "Use atomic_ops.h in libatomic_ops/src" )
+
+  # MacOS has problems with threads and malloc redirect. Disable it for now.
+  # We still use the C++ new/delete etc garbage collection.
+  if (APPLE)
+    append("-Wno-deprecated-declarations" CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
+    add_definitions("-DNO_MARKER_SPECIAL_SIGMASK")
+    set(enable_redirect_malloc OFF CACHE BOOL "Redirect malloc and friends to GC routines.")
+  else()
+    # For any other distribution we redirect all malloc calls in the program.
+    set(enable_redirect_malloc ON CACHE BOOL "Redirect malloc and friends to GC routines.")
```

Comment:
What does this option do? In `gc.cpp` there is our own `malloc` override, but this options suggests that libgc has its own way of doing so.

Reply:
Currently, it is necessary to prevent segmentation faults if configured with thread support. I have not found a configuration yet that does not require it.